src/solver/utils/ortools_utils.cpp

```diff
@@ -354,33 +355,42 @@ std::string availableOrToolsSolversString()
     return solvers.str();
 }
 
-MPSolver* MPSolverFactory(const bool isMip, const std::string& solverName)
+static std::optional<std::string> translateSolverName(const std::string& solverName, bool isMip)
 {
-    MPSolver* solver;
     try
     {
         if (isMip)
         {
-            solver = MPSolver::CreateSolver((OrtoolsUtils::solverMap.at(solverName)).MIPSolverName);
+            return OrtoolsUtils::solverMap.at(solverName).MIPSolverName;
         }
         else
         {
-            solver = MPSolver::CreateSolver((OrtoolsUtils::solverMap.at(solverName)).LPSolverName);
+            return OrtoolsUtils::solverMap.at(solverName).LPSolverName;
         }
+    }
+    catch (const std::out_of_range&)
+    {
+        return {};
+    }
+}
 
-        if (!solver)
-        {
-            std::string msg_to_throw = "Solver " + solverName + " not found. \n";
-            msg_to_throw += "Please make sure that your OR-Tools install supports solver "
-                            + solverName + ".";
+MPSolver* MPSolverFactory(const bool isMip, const std::string& solverName)
+{
+    const std::string notFound = "Solver " + solverName
+                                 + " not found. Please use one of the following "
+                                 + availableOrToolsSolversString();
+    const Antares::Data::AssertionError assertionError(notFound);
 
-            throw Antares::Data::AssertionError(msg_to_throw);
-        }
+    auto internalSolverName = translateSolverName(solverName, isMip);
+    if (!internalSolverName)
+    {
+        throw assertionError;
```

Comment:
shouldn't we throw an std::invalid_argument exception here?

Reply:
That's right